kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/Lifecycles.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+
+/**
+ * The [Lifecycle] for this context, or null if one can't be found.
+ *
+ * @implNote We keep all this very forgiving because we're just using it to keep some logging
```

Comment:
Should drop the `implNote` entirely.

Reply:
It won't show up in rendered kdoc, if that's what you're concerned about.

Reply:
Neat to know, but it's misleading. IIRC we use it for more than just that original reason these days.

Reply:
Removed.